page_analyzer/app.py

```diff
@@ -1,8 +1,115 @@
-from flask import Flask, render_template
+import os
+from typing import Any, Dict
+from urllib.parse import urlparse
+
+import psycopg2
+from dotenv import load_dotenv
+from flask import (
+    Flask,
+    flash,
+    get_flashed_messages,
+    redirect,
+    render_template,
+    request,
+    url_for,
+)
+from validators import url
+
+load_dotenv()
+
 
 app = Flask(__name__)
+app.secret_key = os.getenv("SECRET_KEY")
+DATABASE_URL = os.getenv("DATABASE_URL")
+
+
+def connect_db() -> Any:
+    conn = psycopg2.connect(DATABASE_URL)
+    return conn
+
+
+def get_url_context(url_id: int) -> Dict[str, Any]:
+    conn = connect_db()
+    cur = conn.cursor()
+    cur.execute("SELECT name FROM urls WHERE id='%s';", (url_id,))
+    page_url = cur.fetchone()[0]
+    cur.execute("SELECT created_at FROM urls WHERE id='%s';", (url_id,))
+    created_at = cur.fetchone()[0]
+    cur.close()
+    conn.close()
+    context = {"id": url_id, "name": page_url, "created_at": created_at}
+    return context
+
+
+def validate(url_name: str) -> Dict[str, Any]:
+    errors = {}
+    if not url(url_name) or len(url_name) > 255:
+        errors["name"] = "Некорректный URL"
+    return errors
+
+
+def normalize(url_name: str) -> str:
+    p = urlparse(url_name)
+    normalized = p.scheme + "://" + p.netloc
+    if p.port:
+        port = ":" + str(p.port)
+        normalized = normalized.replace(port, "")
+    return normalized
```

Comment:
wdyt about moving it into a separate module as well? smth like `utils.url` or `utils.validation`